### PR TITLE
HAWQ-917. Refactor feature tests for data type check with new googlet…

### PR DESCRIPTION
**Pull request by paul-guo-**
…est framework

### CODE REVIEW

**Comment by wengyanqing** [comment]
LGTM

File: src/test/feature/type/ans/int8.ans [hunk 1]
```diff
@@ -175,131 +179,131 @@ SELECT '' AS to_char_4, to_char( (q1 * -1), '9999999999999999S'), to_char( (q2 *
...
-            |                123.000
-            |                456.000
-            |   4567890123456789.000
-            |   4567890123456789.000
-            |  -4567890123456789.000
+            | $              456.000
+            | $ 4567890123456789.000
+            | $              123.000
+            | $ 4567890123456789.000
+            | $-4567890123456789.000
```
Comment (xunzhang) [inline src/test/feature/type/ans/int8.ans]: Why did you add `$` sign here?

File: src/test/feature/type/TestType.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#include "gtest/gtest.h"
+
+#include "lib/sql_util.h"
+
+using std::string;
```
Comment (xunzhang) [inline src/test/feature/type/TestType.cpp]: add `#incude <string>`

Comment (paul-guo-) [inline src/test/feature/type/TestType.cpp]: It could compile without this. sql_util.h has included <string>. Do we really need to include this?

**Comment by xunzhang** [comment]
@paul-guo- , LGTM except above comments. Another comments is that according to the folders names defined by @yaoj2 , I suggest you to move your test cases into `catalog` folder instead of independent `type`.

File: src/test/feature/type/ans/int8.ans (hunk 1 above)
Comment (paul-guo-) [inline src/test/feature/type/ans/int8.ans]: I looked into this. I is related to currency info of the locale. You are right this probably need a bit more work.

Comment (paul-guo-) [inline src/test/feature/type/ans/int8.ans]: I'm leaving the to another JIRA HAWQ-925
Set default locale, timezone & datastyle before running sql command/file.

That is for the whole feature test framework.

**Comment by xunzhang** [comment]
+1

**Comment by paul-guo-** [comment]
Merged.
